pinot-query-planner/src/main/java/org/apache/pinot/calcite/sql/fun/PinotOperatorTable.java

```diff
@@ -46,127 +59,296 @@
+      SqlStdOperatorTable.FLOOR,
+      SqlStdOperatorTable.CEIL,
+      SqlStdOperatorTable.TIMESTAMP_ADD,
+      SqlStdOperatorTable.TIMESTAMP_DIFF,
+      SqlStdOperatorTable.CAST,
+
+      SqlStdOperatorTable.EXTRACT,
+      // TODO: The following operators are all rewritten to EXTRACT. Consider removing them because they are all
+      //       supported without rewrite.
+      SqlStdOperatorTable.YEAR,
+      SqlStdOperatorTable.QUARTER,
+      SqlStdOperatorTable.MONTH,
+      SqlStdOperatorTable.WEEK,
+      SqlStdOperatorTable.DAYOFYEAR,
+      SqlStdOperatorTable.DAYOFMONTH,
+      SqlStdOperatorTable.DAYOFWEEK,
+      SqlStdOperatorTable.HOUR,
+      SqlStdOperatorTable.MINUTE,
+      SqlStdOperatorTable.SECOND,
+
+      SqlStdOperatorTable.ITEM,
+      SqlStdOperatorTable.ARRAY_VALUE_CONSTRUCTOR,
+      SqlStdOperatorTable.LISTAGG
+  );
+
+  private static final List<Pair<SqlOperator, List<String>>> STANDARD_OPERATORS_WITH_ALIASES = List.of(
+      Pair.of(SqlStdOperatorTable.CASE, List.of("CASE", "CASE_WHEN")),
+      Pair.of(SqlStdOperatorTable.LN, List.of("LN", "LOG"))
+  );
+
+  /**
+   * This list includes the customized {@link SqlOperator}s.
+   */
+  private static final List<SqlOperator> PINOT_OPERATORS = List.of(
+      // Placeholder for special predicates
+      new PinotSqlFunction("TEXT_MATCH", ReturnTypes.BOOLEAN, OperandTypes.CHARACTER_CHARACTER),
+      new PinotSqlFunction("TEXT_CONTAINS", ReturnTypes.BOOLEAN, OperandTypes.CHARACTER_CHARACTER),
+      new PinotSqlFunction("JSON_MATCH", ReturnTypes.BOOLEAN, OperandTypes.CHARACTER_CHARACTER),
+      new PinotSqlFunction("VECTOR_SIMILARITY", ReturnTypes.BOOLEAN,
+          OperandTypes.family(List.of(SqlTypeFamily.ARRAY, SqlTypeFamily.ARRAY, SqlTypeFamily.INTEGER), i -> i == 2)),
+
+      // Placeholder for special functions to handle MV
+      new PinotSqlFunction("ARRAY_TO_MV", opBinding -> opBinding.getOperandType(0).getComponentType(),
+          OperandTypes.ARRAY),
+
+      // SqlStdOperatorTable.COALESCE without rewrite
+      new SqlFunction("COALESCE", SqlKind.COALESCE,
+          ReturnTypes.LEAST_RESTRICTIVE.andThen(SqlTypeTransforms.LEAST_NULLABLE), null, OperandTypes.SAME_VARIADIC,
+          SqlFunctionCategory.SYSTEM),
+
+      // The scalar function version returns long instead of Timestamp
+      // TODO: Consider unifying the return type to Timestamp
+      new PinotSqlFunction("FROM_DATE_TIME", ReturnTypes.TIMESTAMP_NULLABLE, OperandTypes.family(
+          List.of(SqlTypeFamily.CHARACTER, SqlTypeFamily.CHARACTER, SqlTypeFamily.CHARACTER, SqlTypeFamily.ANY),
+          i -> i > 1))
+  );
+
+  private static final List<Pair<SqlOperator, List<String>>> PINOT_OPERATORS_WITH_ALIASES = List.of(
+  );
+  //@formatter:on
+
+  // Key is canonical name
+  private final Map<String, SqlOperator> _operatorMap;
+  private final List<SqlOperator> _operatorList;
+
+  private PinotOperatorTable() {
+    Map<String, SqlOperator> operatorMap = new HashMap<>();
```

Comment:
It is not clear to me the relation between FunctionRegistry and this class. I would have expected to iterate over all functions declared in FunctionRegistry and add them into calcite, but it looks like we are iterating over another static list. What am I missing?

Reply:
`FunctionRegistry` only contains the scalar function (annotated ones). Added more javadoc to this class